test-utils/src/bitcoin_manager.rs

```diff
@@ -84,24 +105,38 @@ impl<'a> BitcoinManager<'a> {
         address: &Address,
         delay: bool,
     ) -> Result<()> {
-        let client = self.client.get().context("bitcoind not started")?;
         for _ in 0..n_blocks {
             // Sometimes a delay is needed to make the test more realistic which is expected by LDK.
             if delay {
                 tokio::time::sleep(Duration::from_secs(1)).await;
             }
-            client.generate_to_address(1, address).await?;
+            self.client.generate_to_address(1, address).await?;
         }
-        client.wait_for_blockchain_synchronisation().await;
+        self.client.wait_for_blockchain_synchronisation().await;
         Ok(())
     }
 }
 
-pub struct BitcoindCheck(pub Settings);
-
-#[async_trait]
-impl Check for BitcoindCheck {
-    async fn check(&self) -> bool {
-        BitcoindClient::new(&self.0).await.is_ok()
+impl Drop for BitcoinManager<'_> {
```

Comment:
You're duplicating code from the Manager struct. Is the plan to duplicate it four times? Remember we will be adding even more services.

Reply:
Does every service should use the same drop logic?  
Some services run all in memory, some open files, and some we do care.
If the CI/test is happy, I will leave it as it is. 
But the CI failed for several weeks. what do you think?